# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/webrtc/signaling/gtest/mediapipeline_unittest.cpp
# Commit: 786ea95e928b
# Full Hash: 786ea95e928b7bcec53a56b50f1d0ff136c76a62
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 1
# Description:
#   Bug 1654112 - Move NoTrialsConfig into SharedWebrtcState and make WebrtcCallWrapper keep a ref to it. r=ng
#   
#   SharedWebrtcState when managed by PeerConnectionCtx is destroyed on main thread.
#   With interactions with call instances being async, there is no guarantee that
#   the objects owned by SharedWebrtcState (and referenced through raw pointers by
# ==============================================================================

diff -r 2a373d864092 -r 786ea95e928b media/webrtc/signaling/gtest/mediapipeline_unittest.cpp
--- a/media/webrtc/signaling/gtest/mediapipeline_unittest.cpp	Wed Apr 14 17:40:24 2021 +0200
+++ b/media/webrtc/signaling/gtest/mediapipeline_unittest.cpp	Fri Apr 16 14:44:09 2021 +0200
@@ -259,11 +259,10 @@
 
 class TestAgent {
  public:
-  TestAgent(const RefPtr<SharedWebrtcState>& aSharedState,
-            webrtc::WebRtcKeyValueConfig* aTrials)
+  explicit TestAgent(const RefPtr<SharedWebrtcState>& aSharedState)
       : audio_config_(109, "opus", 48000, 2, false),
         call_(WebrtcCallWrapper::Create(mozilla::dom::RTCStatsTimestampMaker(),
-                                        nullptr, aSharedState, aTrials)),
+                                        nullptr, aSharedState)),
         audio_conduit_(
             AudioSessionConduit::Create(call_, test_utils->sts_target())),
         audio_pipeline_(),
@@ -362,9 +361,8 @@
 
 class TestAgentSend : public TestAgent {
  public:
-  TestAgentSend(const RefPtr<SharedWebrtcState>& aSharedState,
-                webrtc::WebRtcKeyValueConfig* aTrials)
-      : TestAgent(aSharedState, aTrials) {
+  explicit TestAgentSend(const RefPtr<SharedWebrtcState>& aSharedState)
+      : TestAgent(aSharedState) {
     using Promise = MozPromise<MediaConduitErrorCode, bool, true>;
     auto rv = WaitFor(InvokeAsync(call_->mCallThread, __func__, [&] {
       return Promise::CreateAndResolve(
@@ -396,9 +394,8 @@
 
 class TestAgentReceive : public TestAgent {
  public:
-  TestAgentReceive(const RefPtr<SharedWebrtcState>& aSharedState,
-                   webrtc::WebRtcKeyValueConfig* aTrials)
-      : TestAgent(aSharedState, aTrials) {
+  explicit TestAgentReceive(const RefPtr<SharedWebrtcState>& aSharedState)
+      : TestAgent(aSharedState) {
     std::vector<AudioCodecConfig> codecs;
     codecs.push_back(audio_config_);
 
@@ -465,10 +462,10 @@
         shared_state_(MakeAndAddRef<SharedWebrtcState>(
             AbstractThread::MainThread(), CreateAudioStateConfig(),
             already_AddRefed(
-                webrtc::CreateBuiltinAudioDecoderFactory().release()))),
-        trials_(new NoTrialsConfig()),
-        p1_(shared_state_, trials_.get()),
-        p2_(shared_state_, trials_.get()) {}
+                webrtc::CreateBuiltinAudioDecoderFactory().release()),
+            WrapUnique(new NoTrialsConfig()))),
+        p1_(shared_state_),
+        p2_(shared_state_) {}
 
   ~MediaPipelineTest() {
     p1_.Shutdown();
@@ -593,7 +590,6 @@
   // we're destroyed.
   UniquePtr<TaskQueueWrapper> main_task_queue_;
   const RefPtr<SharedWebrtcState> shared_state_;
-  const UniquePtr<webrtc::WebRtcKeyValueConfig> trials_;
   TestAgentSend p1_;
   TestAgentReceive p2_;
 };
